## Setting Up the Environment

In [1]:
# Mount Google drive
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [2]:
# Loading libraries
import pandas as pd
import numpy as np

In [3]:
# loading Dataset
cars = pd.read_csv('/content/gdrive/My Drive/CarSample.csv')

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (91) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


## Initial Data Exploration


In [6]:
cars.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62192 entries, 0 to 62191
Columns: 130 entries, Make to Sold_Amount
dtypes: float64(65), int64(6), object(59)
memory usage: 61.7+ MB


The dataset consists of 130 variables, where the target variable `Sold_Amount` is located at the last column.

In [7]:
cars.head(5)

Make      Model MakeCode  ... Age_Comp_Months        KM  Sold_Amount
0  Holden  Commodore     HOLD  ...            93.0  227878.0       2000.0
1  Holden  Commodore     HOLD  ...            86.0  153091.0       6800.0
2  Toyota       RAV4     TOYO  ...            16.0   27374.0      22900.0
3  Holden  Commodore     HOLD  ...            48.0   99452.0      10500.0
4  Toyota     Tarago     TOYO  ...            28.0   44355.0      31320.0

[5 rows x 130 columns]

Taking a glance into the dataframe, we can observe 
- some variables are corresponding to each other eg `MakeCode`corresponds to the indexing code for `Make`, so as `FamilyCode` vs. `Model`.
- variable `Description` are additional info about the car which consists of strings which may not be helpful for model building. The columns are information breakdown of `Description` such as `Series`, `SeriesModelYear`, `BadgeDescription`, `BodyStyleDescription`, `BodyConfigDescription` etc. 
- following variables are `wheelBaseCOnfig`, `Roofline`, `ExtraIdentification` which consists of many missing values. 
- followed by variables describing the drive, gear, engine, fuel, cylinders, electric voltage etc. These variables are related to the performance / condition of a car. 
- The following variables are related to the environmental impact of the car. Eg. `GreenHouseRating`, `AirPollutionRating`, CO2 emmisions, `EmissionStandard` etc. 
- `AncapRating` an indicator of car safety by the standards of [The Australasian New Car Assessment Program](https://www.ancap.com.au/safety-ratings-explained)  are provided. 
- Some [VFACT](https://www.carexpert.com.au/category/vfacts) parameters on class, segment, price are available.

In [ ]:
for col in df.columns:
    print(col)

Make
Model
MakeCode
FamilyCode
YearGroup
MonthGroup
SequenceNum
Description
CurrentRelease
ImportFlag
LimitedEdition
Series
SeriesModelYear
BadgeDescription
BadgeSecondaryDescription
BodyStyleDescription
BodyConfigDescription
WheelBaseConfig
Roofline
ExtraIdentification
DriveDescription
DriveCode
GearTypeDescription
GearLocationDescription
GearNum
DoorNum
EngineSize
EngineDescription
Cylinders
FuelTypeDescription
InductionDescription
OptionCategory
CamDescription
EngineTypeDescription
FuelCapacity
FuelDeliveryDescription
MethodOfDeliveryDescription
GrossCombinationMAss
GrossVehicleMass
VIN
WheelBase
Height
Length
Width
KerbWeight
TareMass
PayLoad
Power
PowerRPMFrom
PowerRPMTo
Torque
TorqueRPMFrom
TorqueRPMTo
RonRating
SeatCapacity
ModelCode
BuildCountryOriginDescription
ValvesCylinder
EngineCycleDescription
EngineConfigurationDescription
EngineLocation
EngineNum
Acceleration
FrontTyreSize
RearTyreSize
FrontRimDesc
RearRimDesc
TowingBrakes
TowingNoBrakes
WarrantyCustAssist
FreeScheduled